doc/developerGuide/pinConfig.rst
####################################################################################
1. State of the I/Os after all the power rails are powered and before configuration?
####################################################################################

SOURCE: https://www.xilinx.com/support/answers/50802.html

The recommended power-on sequence is VCCINT, VCCBRAM, VCCAUX, VCCAUX_IO, and VCCO to achieve minimum current draw and
ensure that the I/Os are 3-stated at power-on.

The 7 series FPGAs contain a pin called PUDC_B. When PUDC_B is Low, internal pull-up resistors are enabled on each
SelectIO pin. When PUDC_B is High, internal pull-up resistors are disabled on each SelectIO pin. The state of this pin
effects the state of the I/O from power-on until configuration completes. Therefore, the I/Os will be 3-stated after
power-on when PUDC is High.

The state of the I/O prior to the rails being powered is not guaranteed.

.. note::

    Due to the presence of the clamp diode, if the I/Os are driven before Vcco powered this will reverse bias the
    Vcco rail. For further information, see (Xilinx Answer 45985).

SOURCE: https://www.xilinx.com/support/documentation/user_guides/ug865-Zynq-7000-Pkg-Pinout.pdf
+----------+----------------+-----------+----------------------------------------------------------+
| Pin Name | Type           | Direction | Description                                              |
|          |                |           | Pull-Up During Configuration (bar)                       |
|          |                |           | Active-Low PUDC_B input enables internal pull-up         |
|          |                |           | resistors on the SelectIO pins after power-up and during |
|          |                |           | configuration.                                           |
| PUDC_B   | Multi-function | Input     |   ° When PUDC_B is Low, internal pull-up resistors       |
|          |                |           |     are enabled on each SelectIO pin.                    |
|          |                |           |   ° When PUDC_B is High, internal pull-up resistors are  |
|          |                |           |     disabled on each SelectIO pin.                       |
|          |                |           | PUDC_B must be tied either directly (or through a 1KΩ or |
|          |                |           | less resistor) to VCCO_34 or GND.                        |
|          |                |           | CAUTION! Do not allow this pin to float before and       |
|          |                |           | during configuration.                                    |
+----------+----------------+-----------+----------------------------------------------------------+

.. note::

    On the Red Pitaya board PUDC_B pin is grounded through a capacitor C100. If needed it can be changed, more 
    information can be found at http://redpitaya.readthedocs.io/en/latest/developerGuide/125-14/shem.html.

################################
2. IO Assignments by Package Pin
################################

A report on current PIN configuration can be generated during synthesis by adding 
'report_io -file $path_out/post_imp_io.rpt' 
to the 
'fpga/red_pitaya_vivado.tcl'
file. This will generate a report in the
'fpga/prj/classic/out/post_imp_io.rpt'


Table bellow shows the current setting of PINs.
+------------+-------------------+------------+-------------------------+---------------+-------------------+---------+------------+------+---------------------+----------------------+---------+------------+-----------+----------+------+------------------+
| Pin Number | Signal Name       | Bank Type  | Pin Name                | Use           | IO Standard       | IO Bank | Drive (mA) | Slew | On-Chip Termination | Off-Chip Termination | Voltage | Constraint | Pull Type | DQS Bias | Vref | Signal Integrity |
+------------+-------------------+------------+-------------------------+---------------+-------------------+---------+------------+------+---------------------+----------------------+---------+------------+-----------+----------+------+------------------+
| A1         | DDR_dm[0]         |            | PS_DDR_DM0_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| A2         | DDR_dq[2]         |            | PS_DDR_DQ2_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| A3         |                   |            | VCCO_DDR_502            | VCCO          |                   |         |            |      |                     |                      |   any** |            |           |          |      |                  |
| A4         | DDR_dq[3]         |            | PS_DDR_DQ3_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| A5         | FIXED_IO_mio[6]   |            | PS_MIO6_500             | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A6         | FIXED_IO_mio[5]   |            | PS_MIO5_500             | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A7         | FIXED_IO_mio[1]   |            | PS_MIO1_500             | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| A8         |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| A9         | FIXED_IO_mio[43]  |            | PS_MIO43_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| A10        | FIXED_IO_mio[37]  |            | PS_MIO37_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A11        | FIXED_IO_mio[36]  |            | PS_MIO36_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A12        | FIXED_IO_mio[34]  |            | PS_MIO34_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A13        |                   |            | VCCO_MIO1_501           | VCCO          |                   |         |            |      |                     |                      |   any** |            |           |          |      |                  |
| A14        | FIXED_IO_mio[32]  |            | PS_MIO32_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A15        | FIXED_IO_mio[26]  |            | PS_MIO26_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A16        | FIXED_IO_mio[24]  |            | PS_MIO24_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A17        | FIXED_IO_mio[20]  |            | PS_MIO20_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A18        |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| A19        | FIXED_IO_mio[16]  |            | PS_MIO16_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| A20        | vinn_i[0]         | High Range | IO_L2N_T0_AD8N_35       | INPUT         | LVCMOS33          |      35 |            |      |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| B1         |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| B2         | DDR_dqs_n[0]      |            | PS_DDR_DQS_N0_502       | BIDIR         | DIFF_SSTL15_T_DCI |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| B3         | DDR_dq[1]         |            | PS_DDR_DQ1_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| B4         | DDR_reset_n       |            | PS_DDR_DRST_B_502       | BIDIR         | SSTL15            |         |            | FAST |                     |            FP_VTT_50 |         | FIXED      |           |          |      | NONE             |
| B5         | FIXED_IO_mio[9]   |            | PS_MIO9_500             | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| B6         |                   |            | VCCO_MIO0_500           | VCCO          |                   |         |            |      |                     |                      |   any** |            |           |          |      |                  |
| B7         | FIXED_IO_mio[4]   |            | PS_MIO4_500             | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| B8         | FIXED_IO_mio[2]   |            | PS_MIO2_500             | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| B9         | FIXED_IO_mio[51]  |            | PS_MIO51_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| B10        | FIXED_IO_ps_srstb |            | PS_SRST_B_501           | BIDIR         | LVCMOS25          |         |         12 | FAST |                     |            FP_VTT_50 |         | FIXED      |           |          |      | NONE             |
| B11        |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| B12        | FIXED_IO_mio[48]  |            | PS_MIO48_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| B13        | FIXED_IO_mio[50]  |            | PS_MIO50_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| B14        | FIXED_IO_mio[47]  |            | PS_MIO47_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| B15        | FIXED_IO_mio[45]  |            | PS_MIO45_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| B16        |                   |            | VCCO_MIO1_501           | VCCO          |                   |         |            |      |                     |                      |   any** |            |           |          |      |                  |
| B17        | FIXED_IO_mio[22]  |            | PS_MIO22_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| B18        | FIXED_IO_mio[18]  |            | PS_MIO18_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| B19        | vinp_i[0]         | High Range | IO_L2P_T0_AD8P_35       | INPUT         | LVCMOS33          |      35 |            |      |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| B20        | vinn_i[1]         | High Range | IO_L1N_T0_AD0N_35       | INPUT         | LVCMOS33          |      35 |            |      |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| C1         | DDR_dq[6]         |            | PS_DDR_DQ6_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| C2         | DDR_dqs_p[0]      |            | PS_DDR_DQS_P0_502       | BIDIR         | DIFF_SSTL15_T_DCI |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| C3         | DDR_dq[0]         |            | PS_DDR_DQ0_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |
| C4         |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| C5         | FIXED_IO_mio[14]  |            | PS_MIO14_500            | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C6         | FIXED_IO_mio[11]  |            | PS_MIO11_500            | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C7         | FIXED_IO_ps_porb  |            | PS_POR_B_500            | BIDIR         | LVCMOS33          |         |         12 | FAST |                     |            FP_VTT_50 |         | FIXED      |           |          |      | NONE             |
| C8         | FIXED_IO_mio[15]  |            | PS_MIO15_500            | BIDIR         | LVCMOS33          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C9         |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| C10        | FIXED_IO_mio[52]  |            | PS_MIO52_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C11        | FIXED_IO_mio[53]  |            | PS_MIO53_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C12        | FIXED_IO_mio[49]  |            | PS_MIO49_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C13        | FIXED_IO_mio[29]  |            | PS_MIO29_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| C14        |                   |            | GND                     | GND           |                   |         |            |      |                     |                      |     0.0 |            |           |          |      |                  |
| C15        | FIXED_IO_mio[30]  |            | PS_MIO30_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| C16        | FIXED_IO_mio[28]  |            | PS_MIO28_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| C17        | FIXED_IO_mio[41]  |            | PS_MIO41_501            | BIDIR         | LVCMOS25          |         |          8 | SLOW |                     |                 NONE |         | FIXED      | PULLUP    |          |      | NONE             |
| C18        | FIXED_IO_mio[39]  |            | PS_MIO39_501            | BIDIR         | LVCMOS25          |         |          8 | FAST |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| C19        |                   | High Range | VCCO_35                 | VCCO          |                   |      35 |            |      |                     |                      |    3.30 |            |           |          |      |                  |
| C20        | vinp_i[1]         | High Range | IO_L1P_T0_AD0P_35       | INPUT         | LVCMOS33          |      35 |            |      |                     |                 NONE |         | FIXED      |           |          |      | NONE             |
| D1         | DDR_dq[5]         |            | PS_DDR_DQ5_502          | BIDIR         | SSTL15_T_DCI      |         |            | FAST |           DCI SPLIT |            FP_VTT_50 |         | FIXED      |           |          |      | SPLIT            |

CHANGES

Changes with RedPitaya 0.92                                      29 May 2014

    Release 0.92 is a bug-fix release, not introducing any new features.
    The bug-fixes resolve mainly the following issues:

    1) Several AUTO button related issues.
    2) Several zoom related issues.
    3) Several Measure panel improvements (refresh rate, readability, accuracy, relevance).
    4) Manual range controls.
    5) Artifacts in signal traces at long time ranges (> 1 s).
    6) Long delays at incremental acquisitions and long time ranges (> 1 s).
    7) Trigger level visibility in plot area based on trigger mode & source.
    8) Trigger level dependency on gain settings.
    9) Arbitrary waveform generator file upload & immediate new waveform activation.
   10) Signal generation on EXT trigger.
   11) Acquire uses correct FPGA filter coefficients.
   12) Problems with double GET/POST requests sent from browser for single click.
   13) Several general SW stability issues, crashing applications at certain conditions.
   14) U-Boot reports correct amount of memory.
   15) Ecosystem fixes:
       - Linux OS built completely from sources, including the ramdisk.
       - Using buildroot for the ramdisk, making it possible to add (almost) any SW.
       - Using nginx dependency libraries from buildroot.
       - Versions embedded to all major Red Pitaya binaries & scripts for traceability.
       - WiFi (wlan0) and wired (eth0) routing conflict when WiFi is used.
   16) Cosmetics: improved logging, unique app icons...
   17) Cleanup of obsolete code.


Changes with RedPitaya 0.90                                      27 Feb 2014

    *) Initial release.



Bazaar/nginx/pcre-8.33/config.h
/* config.h.  Generated from config.h.in by configure.  */
/* config.h.in.  Generated from configure.ac by autoheader.  */


/* PCRE is written in Standard C, but there are a few non-standard things it
can cope with, allowing it to run on SunOS4 and other "close to standard"
systems.

In environments that support the GNU autotools, config.h.in is converted into
config.h by the "configure" script. In environments that use CMake,
config-cmake.in is converted into config.h. If you are going to build PCRE "by
hand" without using "configure" or CMake, you should copy the distributed
config.h.generic to config.h, and edit the macro definitions to be the way you
need them. You must then add -DHAVE_CONFIG_H to all of your compile commands,
so that config.h is included at the start of every source.

Alternatively, you can avoid editing by using -D on the compiler command line
to set the macro values. In this case, you do not have to set -DHAVE_CONFIG_H,
but if you do, default values will be taken from config.h for non-boolean
macros that are not defined on the command line.

Boolean macros such as HAVE_STDLIB_H and SUPPORT_PCRE8 should either be defined
(conventionally to 1) for TRUE, and not defined at all for FALSE. All such
macros are listed as a commented #undef in config.h.generic. Macros such as
MATCH_LIMIT, whose actual value is relevant, have defaults defined, but are
surrounded by #ifndef/#endif lines so that the value can be overridden by -D.

PCRE uses memmove() if HAVE_MEMMOVE is defined; otherwise it uses bcopy() if
HAVE_BCOPY is defined. If your system has neither bcopy() nor memmove(), make
sure both macros are undefined; an emulation function will then be used. */

/* By default, the \R escape sequence matches any Unicode line ending
   character or sequence of characters. If BSR_ANYCRLF is defined (to any
   value), this is changed so that backslash-R matches only CR, LF, or CRLF.
   The build-time default can be overridden by the user of PCRE at runtime. */
/* #undef BSR_ANYCRLF */

/* If you are compiling for a system that uses EBCDIC instead of ASCII
   character codes, define this macro to any value. You must also edit the
   NEWLINE macro below to set a suitable EBCDIC newline, commonly 21 (0x15).
   On systems that can use "configure" or CMake to set EBCDIC, NEWLINE is
   automatically adjusted. When EBCDIC is set, PCRE assumes that all input
   strings are in EBCDIC. If you do not define this macro, PCRE will assume
   input strings are ASCII or UTF-8/16/32 Unicode. It is not possible to build
   a version of PCRE that supports both EBCDIC and UTF-8/16/32. */
/* #undef EBCDIC */

/* In an EBCDIC environment, define this macro to any value to arrange for the
   NL character to be 0x25 instead of the default 0x15. NL plays the role that
   LF does in an ASCII/Unicode environment. The value must also be set in the
   NEWLINE macro below. On systems that can use "configure" or CMake to set
   EBCDIC_NL25, the adjustment of NEWLINE is automatic. */
/* #undef EBCDIC_NL25 */

/* Define to 1 if you have the `bcopy' function. */
#define HAVE_BCOPY 1

/* Define to 1 if you have the <bits/type_traits.h> header file. */
/* #undef HAVE_BITS_TYPE_TRAITS_H */

/* Define to 1 if you have the <bzlib.h> header file. */
/* #undef HAVE_BZLIB_H */

/* Define to 1 if you have the <dirent.h> header file. */
#define HAVE_DIRENT_H 1

/* Define to 1 if you have the <dlfcn.h> header file. */
#define HAVE_DLFCN_H 1

/* Define to 1 if you have the <editline/readline.h> header file. */
/* #undef HAVE_EDITLINE_READLINE_H */

/* Define to 1 if you have the <edit/readline/readline.h> header file. */
/* #undef HAVE_EDIT_READLINE_READLINE_H */

/* Define to 1 if you have the <inttypes.h> header file. */
#define HAVE_INTTYPES_H 1

/* Define to 1 if you have the <limits.h> header file. */
#define HAVE_LIMITS_H 1

/* Define to 1 if the system has the type `long long'. */
#define HAVE_LONG_LONG 1

/* Define to 1 if you have the `memmove' function. */
#define HAVE_MEMMOVE 1

/* Define to 1 if you have the <memory.h> header file. */
#define HAVE_MEMORY_H 1

/* Define if you have POSIX threads libraries and header files. */
/* #undef HAVE_PTHREAD */

/* Have PTHREAD_PRIO_INHERIT. */
/* #undef HAVE_PTHREAD_PRIO_INHERIT */

/* Define to 1 if you have the <readline/history.h> header file. */
/* #undef HAVE_READLINE_HISTORY_H */

/* Define to 1 if you have the <readline/readline.h> header file. */
/* #undef HAVE_READLINE_READLINE_H */

/* Define to 1 if you have the <stdint.h> header file. */
#define HAVE_STDINT_H 1

/* Define to 1 if you have the <stdlib.h> header file. */
#define HAVE_STDLIB_H 1

/* Define to 1 if you have the `strerror' function. */
#define HAVE_STRERROR 1

/* Define to 1 if you have the <string> header file. */
#define HAVE_STRING 1

/* Define to 1 if you have the <strings.h> header file. */
#define HAVE_STRINGS_H 1

/* Define to 1 if you have the <string.h> header file. */
#define HAVE_STRING_H 1

/* Define to 1 if you have `strtoimax'. */
/* #undef HAVE_STRTOIMAX */

/* Define to 1 if you have `strtoll'. */
/* #undef HAVE_STRTOLL */

/* Define to 1 if you have `strtoq'. */
#define HAVE_STRTOQ 1

/* Define to 1 if you have the <sys/stat.h> header file. */
#define HAVE_SYS_STAT_H 1

/* Define to 1 if you have the <sys/types.h> header file. */
#define HAVE_SYS_TYPES_H 1

/* Define to 1 if you have the <type_traits.h> header file. */
/* #undef HAVE_TYPE_TRAITS_H */

/* Define to 1 if you have the <unistd.h> header file. */
#define HAVE_UNISTD_H 1

/* Define to 1 if the system has the type `unsigned long long'. */
#define HAVE_UNSIGNED_LONG_LONG 1

/* Define to 1 if the compiler supports simple visibility declarations. */
#define HAVE_VISIBILITY 1

/* Define to 1 if you have the <windows.h> header file. */
/* #undef HAVE_WINDOWS_H */

/* Define to 1 if you have the <zlib.h> header file. */
/* #undef HAVE_ZLIB_H */

/* Define to 1 if you have `_strtoi64'. */
/* #undef HAVE__STRTOI64 */

/* The value of LINK_SIZE determines the number of bytes used to store links
   as offsets within the compiled regex. The default is 2, which allows for
   compiled patterns up to 64K long. This covers the vast majority of cases.
   However, PCRE can also be compiled to use 3 or 4 bytes instead. This allows
   for longer patterns in extreme cases. */
#define LINK_SIZE 2

/* Define to the sub-directory in which libtool stores uninstalled libraries.
   */
#define LT_OBJDIR ".libs/"

/* The value of MATCH_LIMIT determines the default number of times the
   internal match() function can be called during a single execution of
   pcre_exec(). There is a runtime interface for setting a different limit.
   The limit exists in order to catch runaway regular expressions that take
   for ever to determine that they do not match. The default is set very large
   so that it does not accidentally catch legitimate cases. */
#define MATCH_LIMIT 10000000

/* The above limit applies to all calls of match(), whether or not they
   increase the recursion depth. In some environments it is desirable to limit
   the depth of recursive calls of match() more strictly, in order to restrict
   the maximum amount of stack (or heap, if NO_RECURSE is defined) that is
   used. The value of MATCH_LIMIT_RECURSION applies only to recursive calls of
   match(). To have any useful effect, it must be less than the value of
   MATCH_LIMIT. The default is to use the same value as MATCH_LIMIT. There is
   a runtime method for setting a different limit. */
#define MATCH_LIMIT_RECURSION MATCH_LIMIT

/* This limit is parameterized just in case anybody ever wants to change it.
   Care must be taken if it is increased, because it guards against integer
   overflow caused by enormously large patterns. */
#define MAX_NAME_COUNT 10000

/* This limit is parameterized just in case anybody ever wants to change it.
   Care must be taken if it is increased, because it guards against integer
   overflow caused by enormously large patterns. */
#define MAX_NAME_SIZE 32

/* The value of NEWLINE determines the default newline character sequence.
   PCRE client programs can override this by selecting other values at run
   time. In ASCII environments, the value can be 10 (LF), 13 (CR), or 3338
   (CRLF); in EBCDIC environments the value can be 21 or 37 (LF), 13 (CR), or
   3349 or 3365 (CRLF) because there are two alternative codepoints (0x15 and
   0x25) that are used as the NL line terminator that is equivalent to ASCII
   LF. In both ASCII and EBCDIC environments the value can also be -1 (ANY),
   or -2 (ANYCRLF). */
#define NEWLINE 10

/* Define to 1 if your C compiler doesn't accept -c and -o together. */
/* #undef NO_MINUS_C_MINUS_O */

/* PCRE uses recursive function calls to handle backtracking while matching.
   This can sometimes be a problem on systems that have stacks of limited
   size. Define NO_RECURSE to any value to get a version that doesn't use
   recursion in the match() function; instead it creates its own stack by
   steam using pcre_recurse_malloc() to obtain memory from the heap. For more
   detail, see the comments and other stuff just above the match() function.
   */
/* #undef NO_RECURSE */

/* Name of package */
#define PACKAGE "pcre"

/* Define to the address where bug reports for this package should be sent. */
#define PACKAGE_BUGREPORT ""

/* Define to the full name of this package. */
#define PACKAGE_NAME "PCRE"

/* Define to the full name and version of this package. */
#define PACKAGE_STRING "PCRE 8.33"

/* Define to the one symbol short name of this package. */
#define PACKAGE_TARNAME "pcre"

/* Define to the home page for this package. */
#define PACKAGE_URL ""

/* Define to the version of this package. */
#define PACKAGE_VERSION "8.33"

/* to make a symbol visible */
#define PCRECPP_EXP_DECL extern __attribute__ ((visibility ("default")))

/* to make a symbol visible */
#define PCRECPP_EXP_DEFN __attribute__ ((visibility ("default")))

/* The value of PCREGREP_BUFSIZE determines the size of buffer used by
   pcregrep to hold parts of the file it is searching. This is also the
   minimum value. The actual amount of memory used by pcregrep is three times
   this number, because it allows for the buffering of "before" and "after"
   lines. */
#define PCREGREP_BUFSIZE 20480

/* to make a symbol visible */
#define PCREPOSIX_EXP_DECL extern __attribute__ ((visibility ("default")))

/* to make a symbol visible */
#define PCREPOSIX_EXP_DEFN extern __attribute__ ((visibility ("default")))

/* to make a symbol visible */
#define PCRE_EXP_DATA_DEFN __attribute__ ((visibility ("default")))

/* to make a symbol visible */
#define PCRE_EXP_DECL extern __attribute__ ((visibility ("default")))


/* If you are compiling for a system other than a Unix-like system or
   Win32, and it needs some magic to be inserted before the definition
   of a function that is exported by the library, define this macro to
   contain the relevant magic. If you do not define this macro, a suitable
    __declspec value is used for Windows systems; in other environments
   "extern" is used for a C compiler and "extern C" for a C++ compiler.
   This macro apears at the start of every exported function that is part
   of the external API. It does not appear on functions that are "external"
   in the C sense, but which are internal to the library. */
#define PCRE_EXP_DEFN __attribute__ ((visibility ("default")))

/* Define to any value if linking statically (TODO: make nice with Libtool) */
#define PCRE_STATIC 1

/* When calling PCRE via the POSIX interface, additional working storage is
   required for holding the pointers to capturing substrings because PCRE
   requires three integers per substring, whereas the POSIX interface provides
   only two. If the number of expected substrings is small, the wrapper
   function uses space on the stack, because this is faster than using
   malloc() for each call. The threshold above which the stack is no longer
   used is defined by POSIX_MALLOC_THRESHOLD. */
#define POSIX_MALLOC_THRESHOLD 10

/* Define to necessary symbol if this constant uses a non-standard name on
   your system. */
/* #undef PTHREAD_CREATE_JOINABLE */

/* Define to 1 if you have the ANSI C header files. */
#define STDC_HEADERS 1

/* Define to allow pcretest and pcregrep to be linked with gcov, so that they
   are able to generate code coverage reports. */
/* #undef SUPPORT_GCOV */

/* Define to any value to enable support for Just-In-Time compiling. */
/* #undef SUPPORT_JIT */

/* Define to any value to allow pcregrep to be linked with libbz2, so that it
   is able to handle .bz2 files. */
/* #undef SUPPORT_LIBBZ2 */

/* Define to any value to allow pcretest to be linked with libedit. */
/* #undef SUPPORT_LIBEDIT */

/* Define to any value to allow pcretest to be linked with libreadline. */
/* #undef SUPPORT_LIBREADLINE */

/* Define to any value to allow pcregrep to be linked with libz, so that it is
   able to handle .gz files. */
/* #undef SUPPORT_LIBZ */

/* Define to any value to enable the 16 bit PCRE library. */
/* #undef SUPPORT_PCRE16 */

/* Define to any value to enable the 32 bit PCRE library. */
/* #undef SUPPORT_PCRE32 */

/* Define to any value to enable the 8 bit PCRE library. */
#define SUPPORT_PCRE8 /**/

/* Define to any value to enable JIT support in pcregrep. */
/* #undef SUPPORT_PCREGREP_JIT */

/* Define to any value to enable support for Unicode properties. */
/* #undef SUPPORT_UCP */

/* Define to any value to enable support for the UTF-8/16/32 Unicode encoding.
   This will work even in an EBCDIC environment, but it is incompatible with
   the EBCDIC macro. That is, PCRE can support *either* EBCDIC code *or*
   ASCII/UTF-8/16/32, but not both at once. */
/* #undef SUPPORT_UTF */

/* Define to any value for valgrind support to find invalid memory reads. */
/* #undef SUPPORT_VALGRIND */

/* Version number of package */
#define VERSION "8.33"

/* Define to empty if `const' does not conform to ANSI C. */
/* #undef const */

/* Define to the type of a signed integer type of width exactly 64 bits if
   such a type exists and the standard includes do not define it. */
/* #undef int64_t */

/* Define to `unsigned int' if <sys/types.h> does not define. */
/* #undef size_t */
